### PR TITLE
Expanded the JSON data for delivery_bucket request into nested forms

### PR DESCRIPTION
**Pull request by this-username-is-taken**
Sample data for `/api/delivery_buckets`:

``` javascript
{
  "count": 1,
  "next": null,
  "previous": null,
  "results": [{
    "orders": [{
      "invoice": {
        "total": "6.63",
        "url": "http://127.0.0.1:8000/api/invoices/7/",
        "invoice_num": "7b8eb361-acb3-4626-9ab3-a7a08d3d1ead",
        "status": "POD",
        "payer": "",
        "when_created": "2014-03-04T17:28:15.107Z",
        "when_updated": "2014-03-04T17:28:15.170Z",
        "coupon": null,
        "subtotal": "4.99",
        "tax": "0.65",
        "delivery": "0.99",
        "discount": "0",
        "customer_name": "Test",
        "address": "1234567890",
        "postcode": "L1L1L1",
        "phone": "1234567890",
        "email": "test@test.com",
        "user": "http://127.0.0.1:8000/api/users/2/"
      },
      "status": "WAIT",
      "when_created": "2014-03-04T17:28:15.115Z",
      "when_updated": "2014-03-04T17:28:15.151Z",
      "order_items": [{
        "item": {
          "url": "http://127.0.0.1:8000/api/items/217/",
          "name": "Lotte Milkis Refreshing Milk and Yogurt Falvour 250ml 6pk",
          "category": "http://127.0.0.1:8000/api/categories/30/",
          "description": "",
          "sku": "CCB001",
          "price": "4.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 7,
          "when_added": "2014-02-05T20:04:24.193Z",
          "when_updated": "2014-03-04T17:28:15.143Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/37/",
        "order": "http://127.0.0.1:8000/api/orders/11/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "4.99",
        "item_tax": "0.65"
      }],
      "subtotal": "4.99",
      "comment": ""
    }, {
      "invoice": {
        "total": "138.76",
        "url": "http://127.0.0.1:8000/api/invoices/5/",
        "invoice_num": "acc677c9-7694-4966-97b0-a4224741a2b4",
        "status": "POD",
        "payer": "",
        "when_created": "2014-02-09T21:05:47.080Z",
        "when_updated": "2014-02-10T01:38:42.179Z",
        "coupon": null,
        "subtotal": "121.91",
        "tax": "15.86",
        "delivery": "0.99",
        "discount": "0",
        "customer_name": "bBBBB",
        "address": "100 Westmount Road",
        "postcode": "L1L1L1",
        "phone": "1111111111",
        "email": "a@a.com",
        "user": null
      },
      "status": "WAIT",
      "when_created": "2014-02-09T21:05:47.086Z",
      "when_updated": "2014-02-09T21:05:47.146Z",
      "order_items": [{
        "item": {
          "url": "http://127.0.0.1:8000/api/items/1/",
          "name": "Life Brand no Calorie Sweetener 1 g x 100 Packets",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "Suitable for calorie-reduced diets ",
          "sku": "HPDFDCA004",
          "price": "6.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 4,
          "when_added": "2014-02-05T20:04:13.704Z",
          "when_updated": "2014-02-09T21:06:36.689Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/20/",
        "order": "http://127.0.0.1:8000/api/orders/7/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "6.99",
        "item_tax": "0.91"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/2/",
          "name": "Splenda no Calorie Sweetener Packets 100 Packets",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "Ideal for the whole family ",
          "sku": "HPDFDCA006",
          "price": "6.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 3,
          "when_added": "2014-02-05T20:04:13.720Z",
          "when_updated": "2014-02-09T21:06:36.695Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/21/",
        "order": "http://127.0.0.1:8000/api/orders/7/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "6.99",
        "item_tax": "0.91"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/5/",
          "name": "Webber Naturals Pgx Satisfast Pgx & Whey Protein Drink Mix Rich Chocolate 252 g",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "Helps you feel full for hours. Full of protein. ",
          "sku": "HPDFDCA009",
          "price": "29.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 3,
          "when_added": "2014-02-05T20:04:13.753Z",
          "when_updated": "2014-02-09T21:06:36.704Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/22/",
        "order": "http://127.0.0.1:8000/api/orders/7/",
        "quantity": 2,
        "allow_sub": true,
        "item_cost": "59.98",
        "item_tax": "7.8"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/6/",
          "name": "Bodylogix Herbalean Simple Everyday Weightloss 84 Soft Gels",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "Helps with weightloss. Clinically proven formula ",
          "sku": "HPDFDCA010",
          "price": "29.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 1,
          "when_added": "2014-02-05T20:04:13.764Z",
          "when_updated": "2014-02-09T21:05:47.142Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/23/",
        "order": "http://127.0.0.1:8000/api/orders/7/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "29.99",
        "item_tax": "3.9"
      }],
      "subtotal": "103.95",
      "comment": ""
    }, {
      "invoice": {
        "total": "57.42",
        "url": "http://127.0.0.1:8000/api/invoices/1/",
        "invoice_num": "4ca601db-0416-4ca4-b475-169fab15473d",
        "status": "PEND",
        "payer": "",
        "when_created": "2014-02-05T20:07:55.091Z",
        "when_updated": "2014-02-14T05:04:33.327Z",
        "coupon": null,
        "subtotal": "49.93",
        "tax": "6.5",
        "delivery": "0.99",
        "discount": "0",
        "customer_name": "Vincent",
        "address": "251 lester",
        "postcode": "N2L3W8",
        "phone": "1234567891",
        "email": "a@a.com",
        "user": null
      },
      "status": "PEND",
      "when_created": "2014-02-05T20:07:55.098Z",
      "when_updated": "2014-02-14T05:05:17.963Z",
      "order_items": [{
        "item": {
          "url": "http://127.0.0.1:8000/api/items/1/",
          "name": "Life Brand no Calorie Sweetener 1 g x 100 Packets",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "Suitable for calorie-reduced diets ",
          "sku": "HPDFDCA004",
          "price": "6.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 4,
          "when_added": "2014-02-05T20:04:13.704Z",
          "when_updated": "2014-02-09T21:06:36.689Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/1/",
        "order": "http://127.0.0.1:8000/api/orders/1/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "6.99",
        "item_tax": "0.91"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/2/",
          "name": "Splenda no Calorie Sweetener Packets 100 Packets",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "Ideal for the whole family ",
          "sku": "HPDFDCA006",
          "price": "6.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 3,
          "when_added": "2014-02-05T20:04:13.720Z",
          "when_updated": "2014-02-09T21:06:36.695Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/2/",
        "order": "http://127.0.0.1:8000/api/orders/1/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "6.99",
        "item_tax": "0.91"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/3/",
          "name": "DIUREX Ultra Water Pills 80 Pills",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "For temporary use as a mild dieuretic ",
          "sku": "HPDFDCA007",
          "price": "15.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 1,
          "when_added": "2014-02-05T20:04:13.731Z",
          "when_updated": "2014-02-05T20:07:55.147Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/3/",
        "order": "http://127.0.0.1:8000/api/orders/1/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "15.99",
        "item_tax": "2.08"
      }],
      "subtotal": "29.97",
      "comment": ""
    }, {
      "invoice": {
        "total": "33.01",
        "url": "http://127.0.0.1:8000/api/invoices/3/",
        "invoice_num": "766383cd-e4ae-4577-a8af-9fad82705024",
        "status": "POD",
        "payer": "",
        "when_created": "2014-02-05T20:09:43.127Z",
        "when_updated": "2014-02-05T20:09:43.213Z",
        "coupon": null,
        "subtotal": "28.33",
        "tax": "3.69",
        "delivery": "0.99",
        "discount": "0",
        "customer_name": "AAA",
        "address": "123123123",
        "postcode": "A2A2A2",
        "phone": "1211111111",
        "email": "a@aa.com",
        "user": null
      },
      "status": "WAIT",
      "when_created": "2014-02-05T20:09:43.145Z",
      "when_updated": "2014-02-05T20:09:43.208Z",
      "order_items": [{
        "item": {
          "url": "http://127.0.0.1:8000/api/items/217/",
          "name": "Lotte Milkis Refreshing Milk and Yogurt Falvour 250ml 6pk",
          "category": "http://127.0.0.1:8000/api/categories/30/",
          "description": "",
          "sku": "CCB001",
          "price": "4.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 7,
          "when_added": "2014-02-05T20:04:24.193Z",
          "when_updated": "2014-03-04T17:28:15.143Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/13/",
        "order": "http://127.0.0.1:8000/api/orders/5/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "4.99",
        "item_tax": "0.65"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/218/",
          "name": "Lotte Milkis Melon Falvour 250ml 6pk",
          "category": "http://127.0.0.1:8000/api/categories/30/",
          "description": "",
          "sku": "CCB002",
          "price": "4.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 6,
          "when_added": "2014-02-05T20:04:24.200Z",
          "when_updated": "2014-02-09T21:06:36.728Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/14/",
        "order": "http://127.0.0.1:8000/api/orders/5/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "4.99",
        "item_tax": "0.65"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/225/",
          "name": "3:15PM Original Milk Tea 120g 6pk",
          "category": "http://127.0.0.1:8000/api/categories/30/",
          "description": "",
          "sku": "CCB009",
          "price": "2.29",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 1,
          "when_added": "2014-02-05T20:04:24.283Z",
          "when_updated": "2014-02-05T20:09:43.200Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/15/",
        "order": "http://127.0.0.1:8000/api/orders/5/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "2.29",
        "item_tax": "0.3"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/226/",
          "name": "Nescafe 1+2 17pk",
          "category": "http://127.0.0.1:8000/api/categories/30/",
          "description": "",
          "sku": "CCB010",
          "price": "2.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 2,
          "when_added": "2014-02-05T20:04:24.288Z",
          "when_updated": "2014-02-09T21:05:47.161Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/16/",
        "order": "http://127.0.0.1:8000/api/orders/5/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "2.99",
        "item_tax": "0.39"
      }],
      "subtotal": "15.26",
      "comment": ""
    }, {
      "invoice": {
        "total": "142.59",
        "url": "http://127.0.0.1:8000/api/invoices/6/",
        "invoice_num": "6c6d555a-2d04-43cb-9af8-b0c9588797d2",
        "status": "POD",
        "payer": "",
        "when_created": "2014-02-09T21:06:36.631Z",
        "when_updated": "2014-02-09T21:06:36.747Z",
        "coupon": null,
        "subtotal": "125.3",
        "tax": "16.3",
        "delivery": "0.99",
        "discount": "0",
        "customer_name": "CCCC",
        "address": "111111111",
        "postcode": "A1A1A1",
        "phone": "1111111111",
        "email": "a@a.com",
        "user": null
      },
      "status": "WAIT",
      "when_created": "2014-02-09T21:06:36.656Z",
      "when_updated": "2014-02-09T21:06:36.742Z",
      "order_items": [{
        "item": {
          "url": "http://127.0.0.1:8000/api/items/217/",
          "name": "Lotte Milkis Refreshing Milk and Yogurt Falvour 250ml 6pk",
          "category": "http://127.0.0.1:8000/api/categories/30/",
          "description": "",
          "sku": "CCB001",
          "price": "4.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 7,
          "when_added": "2014-02-05T20:04:24.193Z",
          "when_updated": "2014-03-04T17:28:15.143Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/33/",
        "order": "http://127.0.0.1:8000/api/orders/10/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "4.99",
        "item_tax": "0.65"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/218/",
          "name": "Lotte Milkis Melon Falvour 250ml 6pk",
          "category": "http://127.0.0.1:8000/api/categories/30/",
          "description": "",
          "sku": "CCB002",
          "price": "4.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 6,
          "when_added": "2014-02-05T20:04:24.200Z",
          "when_updated": "2014-02-09T21:06:36.728Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/34/",
        "order": "http://127.0.0.1:8000/api/orders/10/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "4.99",
        "item_tax": "0.65"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/219/",
          "name": "Lotte Milkis Strawberry Falvour 250ml 6pk",
          "category": "http://127.0.0.1:8000/api/categories/30/",
          "description": "",
          "sku": "CCB003",
          "price": "4.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 2,
          "when_added": "2014-02-05T20:04:24.237Z",
          "when_updated": "2014-02-09T21:06:36.733Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/35/",
        "order": "http://127.0.0.1:8000/api/orders/10/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "4.99",
        "item_tax": "0.65"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/224/",
          "name": "3:15PM Roasted Milk Tea 120g 6pk",
          "category": "http://127.0.0.1:8000/api/categories/30/",
          "description": "",
          "sku": "CCB008",
          "price": "2.29",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 1,
          "when_added": "2014-02-05T20:04:24.278Z",
          "when_updated": "2014-02-09T21:06:36.738Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/36/",
        "order": "http://127.0.0.1:8000/api/orders/10/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "2.29",
        "item_tax": "0.3"
      }],
      "subtotal": "17.26",
      "comment": ""
    }, {
      "invoice": {
        "total": "142.59",
        "url": "http://127.0.0.1:8000/api/invoices/6/",
        "invoice_num": "6c6d555a-2d04-43cb-9af8-b0c9588797d2",
        "status": "POD",
        "payer": "",
        "when_created": "2014-02-09T21:06:36.631Z",
        "when_updated": "2014-02-09T21:06:36.747Z",
        "coupon": null,
        "subtotal": "125.3",
        "tax": "16.3",
        "delivery": "0.99",
        "discount": "0",
        "customer_name": "CCCC",
        "address": "111111111",
        "postcode": "A1A1A1",
        "phone": "1111111111",
        "email": "a@a.com",
        "user": null
      },
      "status": "PEND",
      "when_created": "2014-02-09T21:06:36.638Z",
      "when_updated": "2014-02-12T22:57:34.037Z",
      "order_items": [{
        "item": {
          "url": "http://127.0.0.1:8000/api/items/1/",
          "name": "Life Brand no Calorie Sweetener 1 g x 100 Packets",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "Suitable for calorie-reduced diets ",
          "sku": "HPDFDCA004",
          "price": "6.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 4,
          "when_added": "2014-02-05T20:04:13.704Z",
          "when_updated": "2014-02-09T21:06:36.689Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/27/",
        "order": "http://127.0.0.1:8000/api/orders/9/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "6.99",
        "item_tax": "0.91"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/2/",
          "name": "Splenda no Calorie Sweetener Packets 100 Packets",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "Ideal for the whole family ",
          "sku": "HPDFDCA006",
          "price": "6.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 3,
          "when_added": "2014-02-05T20:04:13.720Z",
          "when_updated": "2014-02-09T21:06:36.695Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/28/",
        "order": "http://127.0.0.1:8000/api/orders/9/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "6.99",
        "item_tax": "0.91"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/4/",
          "name": "Lean+ Weight Loss Support 120 Capsules",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "To increase metabolism and help maintain healthy weight ",
          "sku": "HPDFDCA008",
          "price": "57.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 1,
          "when_added": "2014-02-05T20:04:13.742Z",
          "when_updated": "2014-02-09T21:06:36.699Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/29/",
        "order": "http://127.0.0.1:8000/api/orders/9/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "57.99",
        "item_tax": "7.54"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/5/",
          "name": "Webber Naturals Pgx Satisfast Pgx & Whey Protein Drink Mix Rich Chocolate 252 g",
          "category": "http://127.0.0.1:8000/api/categories/3/",
          "description": "Helps you feel full for hours. Full of protein. ",
          "sku": "HPDFDCA009",
          "price": "29.99",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 3,
          "when_added": "2014-02-05T20:04:13.753Z",
          "when_updated": "2014-02-09T21:06:36.704Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/30/",
        "order": "http://127.0.0.1:8000/api/orders/9/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "29.99",
        "item_tax": "3.9"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/8/",
          "name": "Oh Yeah! Source of Energy Chocolate & Caramel Flavour 45 g",
          "category": "http://127.0.0.1:8000/api/categories/4/",
          "description": "High in protein, source of energy ",
          "sku": "HPDFDPB002",
          "price": "2.79",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 2,
          "when_added": "2014-02-05T20:04:13.787Z",
          "when_updated": "2014-02-09T21:06:36.709Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/31/",
        "order": "http://127.0.0.1:8000/api/orders/9/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "2.79",
        "item_tax": "0.36"
      }, {
        "item": {
          "url": "http://127.0.0.1:8000/api/items/9/",
          "name": "Isofemme High Protein Energy Bar Chocolate Decadent Almond Delight",
          "category": "http://127.0.0.1:8000/api/categories/4/",
          "description": "Conquers cravings in between meals ",
          "sku": "HPDFDPB003",
          "price": "3.29",
          "sales_price": "0",
          "tax_class": "0.13",
          "out_of_stock": false,
          "on_sale": false,
          "max_quantity_per_order": 0,
          "featured": "",
          "sold_number": 2,
          "when_added": "2014-02-05T20:04:13.797Z",
          "when_updated": "2014-02-09T21:06:36.714Z"
        },
        "url": "http://127.0.0.1:8000/api/order_items/32/",
        "order": "http://127.0.0.1:8000/api/orders/9/",
        "quantity": 1,
        "allow_sub": true,
        "item_cost": "3.29",
        "item_tax": "0.43"
      }],
      "subtotal": "108.04",
      "comment": ""
    }],
    "assignor": {
      "id": 1,
      "url": "http://127.0.0.1:8000/api/users/1/",
      "username": "wensi",
      "first_name": "",
      "last_name": "",
      "email": "wensi@live.com",
      "last_login": "2014-03-18T20:27:34.549Z",
      "date_joined": "2014-02-05T20:06:00Z"
    },
    "assignee": {
      "id": 1,
      "url": "http://127.0.0.1:8000/api/users/1/",
      "username": "wensi",
      "first_name": "",
      "last_name": "",
      "email": "wensi@live.com",
      "last_login": "2014-03-18T20:27:34.549Z",
      "date_joined": "2014-02-05T20:06:00Z"
    },
    "url": "http://127.0.0.1:8000/api/delivery_bucket/1/",
    "start": "2014-03-18T20:32:52Z",
    "end": "2014-03-18T20:32:54Z"
  }]
}
```

### CODE REVIEW

File: api/serializers.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ class Meta:
     model = Invoice
 
 class OrderSerializer(serializers.HyperlinkedModelSerializer):
+  order_items = serializers.HyperlinkedRelatedField(many=True, read_only=True, view_name='orderitem-detail')
```
Comment (gregwym) [inline api/serializers.py]: We don't really need to expose `order_items` as a standalone API entry. So let's remove it from the router, delete that view set and point this `order_items` field (in `OrderSerializer`) to `OrderItemSerializer`.

File: api/serializers.py [hunk 2]
```diff
@@ -52,6 +53,27 @@ class Meta:
...
+  class Meta:
+    model = DeliveryWindow
+
+class DeliveryBucketOrderSerializer(serializers.HyperlinkedModelSerializer):
+  order_items = DeliveryBucketOrderItemSerializer()
+  invoice = InvoiceSerializer()
+  delivery_window = DeliveryBucketDeliveryWindowSerializer()
+  class Meta:
+    model = Order
+    fields = ['invoice', 'status', 'when_created', 'when_updated', 'order_items', 'subtotal', 'comment', 'delivery_window']
```
Comment (gregwym) [inline api/serializers.py]: We should use `DeliveryBucketOrder` as the model.

File: api/views.py [hunk 3]
```diff
@@ -72,4 +72,5 @@ class CouponViewSet(viewsets.ReadOnlyModelViewSet):
 class DeliveryBucketViewSet(viewsets.ReadOnlyModelViewSet):
     queryset = DeliveryBucket.objects.all()
     serializer_class = DeliveryBucketSerializer
+    filter_fields = ['assignee__username']
```
Comment (gregwym) [inline api/views.py]: why not `assignee__id`?

Comment (this-username-is-taken) [inline api/views.py]: I think it's easier to remember than userid.

File: api/serializers.py (hunk 2 above)
Comment (this-username-is-taken) [inline api/serializers.py]: I don't understand what you mean. We don't have a `DeliveryBucketOrder`

File: api/views.py (hunk 3 above)
Comment (gregwym) [inline api/views.py]: I guess API isn't for human, and primary key is generally faster to look up.

File: api/serializers.py (hunk 2 above)
Comment (gregwym) [inline api/serializers.py]: https://github.com/Fruitex/fruitex/blob/dev/delivery/models.py#L4

Comment (this-username-is-taken) [inline api/serializers.py]: Why would you want another wrapper?

Comment (gregwym) [inline api/serializers.py]: What you mean by another wrapper?

Comment (this-username-is-taken) [inline api/serializers.py]: Isn't `DeliveryBucketOrder` just a wrapper over `Order`?

Comment (gregwym) [inline api/serializers.py]: No, it is for the relationship between bucket and order, not a wrapper. In the future, we will store anything about this relationship in this model. For example, the route.

**Pull request merged**